Add fate tests for palette act and field

QSXW opened the pull request:
The fix for DUAL_CHROMA_TREE coding unit was lost in our rebased commits. Please help review https://github.com/ffvvc/FFmpeg/pull/291/commits/0f9b47af108fd39a77970f5d871f781ea9da6541

nuomi2021 committed:
add github workflow

nowrep committed:
vulkan_encode_h264/5: Fix uninitialized return value in write_extra_headers

QSXW committed:
avcodec/vvc/ctu: should use the width and height of the start component

This commit fixed decoding the DUAL_TREE_CHROMA palette coding unit

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

QSXW committed:
fate/vvc: add vvc-conformance-10b422_L_5

This commit added 10b422_L_5 for testing palette mode.

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

QSXW committed:
fate/vvc: add vvc-conformance-ACT_A_3

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

QSXW committed:
fate/vvc: add vvc-conformance-FIELD_A_4

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

QSXW committed:
avcodec/h274: fix hash verification on BE

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

nuomi2021 commented:
> The fix for DUAL_CHROMA_TREE coding unit was lost in our rebased commits. Please help review [0f9b47a](https://github.com/ffvvc/FFmpeg/commit/0f9b47af108fd39a77970f5d871f781ea9da6541)

thank you for the patch

For which specific clip are you seeing the issue? All ACT clips pass locally with the upstream code.

like:
ACTPIC_A_Huawei_3.bit  ACTPIC_B_Huawei_3.bit  ACTPIC_C_Huawei_3.bit  ACT_A_Kwai_3.bit       ACT_B_Kwai_3.bit

QSXW commented:
> > The fix for DUAL_CHROMA_TREE coding unit was lost in our rebased commits. Please help review [0f9b47a](https://github.com/ffvvc/FFmpeg/commit/0f9b47af108fd39a77970f5d871f781ea9da6541)
> 
> thank you for the patch
> 
> For which specific clip are you seeing the issue? All ACT clips pass locally with the upstream code.
> 
> like: ACTPIC_A_Huawei_3.bit ACTPIC_B_Huawei_3.bit ACTPIC_C_Huawei_3.bit ACT_A_Kwai_3.bit ACT_B_Kwai_3.bit

It's the one encoded by myself. See https://github.com/ffvvc/FFmpeg/pull/224#discussion_r2033909419

We should use the width and height of the start component. When the cu is DUAL_CHROMA_TREE, the start component is the first transform block, so we need to use the tb width and height.

QSXW committed:
avcodec/vvc/dec: fix typo and also output log when the checksum is correct

It's helpful for developers and the same as the hevcdec.

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

nuomi2021 commented:
> It's the one encoded by myself. See [#224 (comment)](https://github.com/ffvvc/FFmpeg/pull/224#discussion_r2033909419)
> 
> We should use the width and height of the start component. When the cu is DUAL_CHROMA_TREE, the start component is the first transform block, so we need to use the tb width and height.

could you share the DUAL_CHROMA_TREE  clip with me? 
Thank you

nuomi2021 commented:
It's better to remove the company name from the clip, just like the other FATE clips.

QSXW commented:
> > It's the one encoded by myself. See [#224 (comment)](https://github.com/ffvvc/FFmpeg/pull/224#discussion_r2033909419)
> > We should use the width and height of the start component. When the cu is DUAL_CHROMA_TREE, the start component is the first transform block, so we need to use the tb width and height.
> 
> could you share the DUAL_CHROMA_TREE clip with me? Thank you

Sure. https://github.com/user-attachments/assets/91fce515-47bf-45e6-bc32-4473eba2260f

To upload the clip, I changed the file extension from `bit` to `mp4`. You need to change it back.

Open the link and use `Ctrl + S` to download the clip.

nuomi2021 commented on libavcodec/vvc/ctu.c:
```diff
@@ -2039,10 +2041,11 @@ static int palette_subblock_data(VVCLocalContext *lc,
         const int scale    = ff_vvc_palette_derive_scale(lc, tu, tb);
-        const int hs       = sps->hshift[c];
-        const int vs       = sps->vshift[c];
+        const int hs       = tu->tbs[0].c_idx ? 0 : sps->hshift[c_idx];
```
How about defining a start_index? Then this can be simplified to sps->hshift[c_idx] - sps->hshift[start_index], which makes it more readable.

QSXW replied:
Sure.

nuomi2021 commented on libavcodec/h274.c:
```diff
@@ -932,10 +959,22 @@ int ff_h274_hash_verify(H274HashContext *c, const H274SEIPictureHash *hash,
+#if HAVE_BIGENDIAN
+        if (c->type != HASH_CHECKSUM) {
+            if (ps && !c->buf) {
+                av_fast_malloc(&c->buf, &c->buf_size,
```
do we need to free this?

QSXW replied:
Yes update.

nuomi2021 commented on libavcodec/h274.c:
```diff
@@ -932,10 +959,22 @@ int ff_h274_hash_verify(H274HashContext *c, const H274SEIPictureHash *hash,
+#if HAVE_BIGENDIAN
+        if (c->type != HASH_CHECKSUM) {
+            if (ps && !c->buf) {
+                av_fast_malloc(&c->buf, &c->buf_size,
```
what if the frame resolution or ps changed?

QSXW replied:
updated.

russelltg committed:
hwcontext_vulkan: fix exporting multi-plane DRM modifiers

Previously, it was assumed that `drmFormatModifierPlaneCount` was one
for all modifiers when exporting, which is not always the case, in
particular for AMD GPUs and maybe others.

Fetch the number of memory planes and fill the structs appropriately in this situation.

The encoded stream is still bad in the case whre modifers are involved,
but I think this patch still stands on its own and I suspect that may be a driver bug.

A potential improvement that could be make is to cache the format
information, so we can avoid the two GetPhysicalDeviceFormatProperties2
calls for each export, as well as the allocation. I doubt this is very
expensive, but seemed worth noting.

v2 changes: query the format properties with the test image created in
`vulkan_frames_init` to avoid allocating space for the query during
export

Signed-off-by: Russell Greene <russellgreene8@gmail.com>

cyanreg committed:
lavc/vulkan/common: sign-ify lengths

This makes left_bits return useful data rather than overflowing, and
also saves some 64-bit integer operations, which is still always a plus sadly.

cyanreg committed:
vf_libplacebo: add support for specifying a LUT for the input

This makes it possible to apply Adobe .cube files to inputs.

cyanreg committed:
hwcontext_vulkan: temporarily disable host_image_copy

NVIDIA's support for it is a disaster.
Of no benefit to other vendors.

NVIDIA are working on fixing it, but it may take time.

cyanreg committed:
hwcontext_vulkan: enable uniformBufferStandardLayout

pz9115 committed:
configure: don't disable '-ftree-vectorize' on GCC >= 13 on major architectures

This changes configure to stop disabling -ftree-vectorize on
GCC versions 13 and newer, on major architectures.

Background:
- Original `-fno-tree-vectorize` was added in 2009 in commit
  973859f5230e to avoid compiler errors.
- Re-enabled in 2016 in commit cb8646af24bd but caused failures due
  to inline CABAC assembly issues and was disabled again in
  fd6dbc53855fb.
- Commit 182663a58a7a in 2023 fixed the inline CABAC assembly issues.
- Recent versions of GCC, in particular 13 and newer, seem to
  generally work reliably with respect to vectorization, although bugs
  have been observed on Loongarch.

Cautiously allow the GCC default of having vectorization enabled,
on major architectures where we expect to see enough testing. If
further issues are observed, they should be reported and noted here in
configure, so the workarounds can be scoped and version limited.

kasper93 committed:
configure: treat unrecognized flags as errors on MSVC

This is important for feature checking to work correctly.

It can happen that an unrecognized flag passes the compile test with
only a warning, while failing in preprocessor-only check with an error.
This causes all test_cpp calls to fail and silently produces arguably
broken MSVC builds. Also, all check_* functions don't work as expected,
because they assume the check passed, even though there was a warning.

Additionally, this brings the behavior in line with GCC/Clang based
builds, failing early on unrecognized flags instead of silently
continuing with warnings in the log.

The /options:strict option is available starting in Visual Studio 2022
version 17.0. Because of that, we cannot use check_cflags alone, as it
would add this flag for older MSVC versions and produce warnings. So, we
need to manually perform a version check. A bit of a chicken and egg
problem.

Perform the version check before adding extra flags from the user to
ensure we don't silently fail the preprocessor check due to invalid
flags on older MSVC versions. Note that behavior differs depending on
whether we are compiling or only preprocessing.

This fixes silent different between handling:

`cl.exe -P foo c.c`
    c1: fatal error C1083: Cannot open source file: 'foo': No such file
    or directory

`cl.exe -c foo c.c`
    cl : Command line warning D9024 : unrecognized source file type
    'foo', object file assumed

Where -P fails, while -c throws warnings only. Of course `foo` is
completely bogus here, but depends on the flags or configuration this
may be unsupported argument. Or even some converted path from MSYS when
run inside it. The objective is to always error out instead of silently
hiding this.

Use check_cflags even after the _MSC_FULL_VER check, for non-MSVC
compilers. For example Clang-CL impersonate MSVC, but does not support
-options:strict flag currently.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
configure: filter link flags separately for MSVC

This avoids adding flags that cl.exe doesn't understand.

Fixes cases where external libraries pkg-config file adds `-L` to the
cflags, strip it before passing to cl.exe.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
configure: don't use dangling temp file state

There was implicit assumption that the $TMPC file is empty when doing
--cpu=host checks. This breaks if any check is done before that.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
avformat/ffmetadec: don't compare undefined string

Fixes use-of-uninitialized-value when bp.len == 0.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
avcodec/dsicinaudio: set missing sample rate

Commit f566032bfda2fee4fb6388d5906d1957a1ed868a added frame validation.
Since then this decoder has been failing validation of sample rate
value.

Found by OSS-Fuzz.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
avcodec/evrcdec: set missing sample rate

Commit f566032bfda2fee4fb6388d5906d1957a1ed868a added frame validation.
Since then this decoder has been failing validation of sample rate
value.

Found by OSS-Fuzz.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
avcodec/qcelpdec: set missing sample rate

Commit f566032bfda2fee4fb6388d5906d1957a1ed868a added frame validation.
Since then this decoder has been failing validation of sample rate
value.

Found by OSS-Fuzz.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
avcodec/qcelpdec: set missing sample rate

Commit f566032bfda2fee4fb6388d5906d1957a1ed868a added frame validation.
Since then this decoder has been failing validation of sample rate
value.

Found by OSS-Fuzz.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

kasper93 committed:
tools/target_fuzzer: don't spam stderr

Printing dummy logs during fuzzing can significantly slow the process
and blow the size of logs, making them both unredable and huge.

Keep the loggging commented-out for easy local restore if needed.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

jamrial committed:
avformat/mov: set primary extradata based on the first Sample only if it's not already in place

If the first Sample references the first stsd entry, then setting it here is
redundant.

Signed-off-by: James Almer <jamrial@gmail.com>

kasper93 committed:
forgejo/workflows: sort file list for cache hash

Makes find output consistent.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>

michaelni committed:
avcodec/motion_est: Remove double protection from init_ref()

Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>

QSXW commented:
@nuomi2021 Let's get this to mainline. Should I send it to the mailing list?

jamrial committed:
avformat/iff: fix EOF check

The check to return on EOF should not be inside a block that will not be entered after reaching EOF.
Should fix "libavcodec/bytestream.h:144:27: runtime error: applying zero offset to null pointer".

Signed-off-by: James Almer <jamrial@gmail.com>

T-bagwell committed:
tests/fate/hlsenc: add testcase of hls fragment mp4 named cmfa

Add allow extension name cmfa and cmfv test, this testcase only
cover fragment mp4 named cmfa.

ticket description in ticket/11526

QSXW committed:
avformat/Makefile: fix error unresolved external symbol ff_dtls_protocol

Introduced by 307983b292adb60c82e75c8e5e41deedc89efbf0

Use the following command line to reproduce the issue:
./configure --toolchain=msvc --disable-asm --enable-ffmpeg \
--disable-everything --enable-decoder=vvc --enable-parser=vvc \
--enable-demuxer='vvc,mpegts' --enable-protocol='file,pipe' \
--enable-encoder='rawvideo,wrapped_avframe' \
--enable-muxer='rawvideo,md5,null'

Signed-off-by: Wu Jianhua <toqsxw@outlook.com>

michaelni committed:
.forgejo/CODEOWNERS: add myself to a few things

Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>

GyanD committed:
forgejo/CODEOWNERS: add myself to docs

nuomi2021 commented:
+1. Better to group patches with related functions and send them to the mailing list separately.
